core/workflows/workflow_base.py

```
@@ -0,0 +1,196 @@
+
+        checkpoint_path = self.config.staging_path / f"{self.config.name}_checkpoint.json"
+
+        if not checkpoint_path.exists():
+            return None
+
+        try:
+            import json
+            with open(checkpoint_path, 'r') as f:
+                data = json.load(f)
+            logger.info(f"Checkpoint loaded from {checkpoint_path}")
+            return data
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)